### PR TITLE
Bump Smallrye Reactive Messaging version from 4.12.0 to 4.13.0

### PR DESCRIPTION
**Pull request by ozangunalp**
- Micrometer Messaging observability API implementation
- Kafka Request Reply support with serde auto detection
- Documentation updates new features

### CODE REVIEW

File: docs/src/main/asciidoc/kafka.adoc [hunk 1]
```diff
@@ -497,6 +497,9 @@ The partitions of the subscribed topic will be divided among the consumer thread
 Note that if the `partitions` value exceed the number of partitions of the topic, some consumer threads won't be assigned any partitions.
 +
```
Comment (cescoffier) [inline docs/src/main/asciidoc/kafka.adoc:498]: I would add a note that `partitions` is deprecated.

File: docs/src/main/asciidoc/kafka.adoc [hunk 2]
```diff
@@ -658,6 +661,36 @@ mp.messaging.incoming.your-channel.group.id=${quarkus.uuid}
...
+
+assigns the consumer to:
+
+- Partition 0 of topic 'topic1', setting the initial position at offset 10.
+- Partition 1 of topic 'topic2', setting the initial position at offset 20.
+
+The topic, partition, and offset in each triplet can have the following variations:
+
+- If the topic is omitted, the configured topic will be used.
+- If the offset is omitted, partitions are assigned to the consumer but won't be seeked to offset.
```
Comment (cescoffier) [inline docs/src/main/asciidoc/kafka.adoc:689]:
```suggestion
- If the offset is omitted, partitions are assigned to the consumer but won't be sought to offset.
```

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/kafka.adoc (hunk 2 above)
Comment (ozangunalp) [reply docs/src/main/asciidoc/kafka.adoc]: Nice catch!

File: docs/src/main/asciidoc/kafka.adoc (hunk 1 above)
Comment (ozangunalp) [reply docs/src/main/asciidoc/kafka.adoc:498]: I mentioned the `concurrency` and added a deprecation note for `partitions`.

**Comment by ozangunalp** [comment]
Rebased to main

**Review by cescoffier — APPROVED** [review]

**Pull request merged**